### PR TITLE
feat(auth-manager): added token generation

### PR DESCRIPTION
**Pull request by CptSchnitz**
Added feature to create token automatically for connections.
When token string is empty in the request, and a private key is defined for the environment a new token will be generated.

### CODE REVIEW

File: packages/auth-manager/README.md [hunk 1]
```diff
@@ -1,11 +1,6 @@
...
 
-## Usage
-
-```
-const authManager = require('auth-manager');
-
-// TODO: DEMONSTRATE API
-```
+When creating or updating a connection, a token will be generated automatically based on the private key.
+If for any reason the token generation failed, or no private key existed, the token field will be an empty string, and the reason will be logged.
```
Comment (shimoncohen) [inline packages/auth-manager/README.md:6]: `existed` -> `exists`

**Review by shimoncohen — APPROVED** [review]

**Comment by shimoncohen** [comment]
@AnguIar Could you please look at this review as well?

File: packages/auth-manager/src/connection/models/connectionManager.ts [hunk 2]
```diff
@@ -89,4 +95,27 @@ export class ConnectionManager {
...
+      });
+      return '';
+    }
+
+    try {
+      return await generateToken(key.privateKey as JWK, connection.name, key.privateKey.kid);
+    } catch (error) {
+      this.logger.error({ msg: 'could not generate token', error });
+      return '';
+    }
```
Comment (syncush) [inline packages/auth-manager/src/connection/models/connectionManager.ts:119]: I don't like the best-effort approach, better having a flag specifying that token is not a must.

**Review by syncush — CHANGES_REQUESTED** [review]

File: packages/auth-manager/tests/unit/connection/models/connectionManager.spec.ts [hunk 3]
```diff
@@ -114,7 +119,19 @@ describe('ConnectionManager', () => {
         return fn({
           withRepository: jest
             .fn()
-            .mockImplementation((callValue) => (callValue === connectionRepo ? connectionTransactionRepo : domainTransactionRepo)),
+            // .mockImplementation((callValue) => (callValue === connectionRepo ? connectionTransactionRepo : domainTransactionRepo)),
```
Comment (shimoncohen) [inline packages/auth-manager/tests/unit/connection/models/connectionManager.spec.ts:122]: Please remove comment.

**Review by shimoncohen — APPROVED** [review]

File: packages/auth-manager/tests/unit/connection/models/connectionManager.spec.ts (hunk 3 above)
Comment (CptSchnitz) [reply packages/auth-manager/tests/unit/connection/models/connectionManager.spec.ts]: removed

File: packages/auth-manager/src/connection/models/connectionManager.ts (hunk 2 above)
Comment (CptSchnitz) [reply packages/auth-manager/src/connection/models/connectionManager.ts]: Added a query param to specify the requested behaviour.

**Review by syncush — APPROVED** [review]

**Review by shimoncohen — APPROVED** [review]

**Pull request merged**